plugins/newsdownloader.koplugin/lib/xml.lua

```diff
@@ -371,6 +371,14 @@ local xmlParser = function(handler)
                       ["&amp;"] = "&",
                       ["&quot;"] = '"',
                       ["&apos;"] = "'",
+                      ["&#8210;"] = "-",
+                      ["&#x2012;"] = "-",
+                      ["&#8211;"] = "-",
+                      ["&#x2013;"] = "-",
+                      ["&#8212;"] = "-",
+                      ["&#x2014;"] = "-",
+                      ["&#8213;"] = "-",
+                      ["&#x2015;"] = "-",
                       ["&#(%d+);"] = function (x)
```

Comment:
Btw, I think it might be better (as in clearer) to replace this with a reference to our utility function where it's used so that the library remains vanilla.